### DISCUSSION TITLE
TEXT_DEHYPHENATE in `page.search_for`

### DISCUSSION THREAD

**Discussion by AnweshGangula** [pymupdf/PyMuPDF#1281] [Q&A] [answered]
The [documentation](https://pymupdf.readthedocs.io/en/latest/page.html#Page.search_for) show that `page.search_for` mentions a flag "TEXT_DEHYPHENATE". Can you help me understand that?
The [TEXT_DEHYPHENATE documentation](https://pymupdf.readthedocs.io/en/latest/vars.html#TEXT_DEHYPHENATE) says that it ignores hyphens at line ends. But adding this flag doesn't seem to make a difference in `page.search_for`.

This is how I have added it:
```python
page.search_for("needle",  flags=fitz.TEXT_DEHYPHENATE)
```

**Comment by JorjMcKie** [comment, marked as answer]
Search already has this flag set! This shows the method definition:
``search_for(needle, clip=clip, quads=False, flags=TEXT_DEHYPHENATE | TEXT_PRESERVE_WHITESPACE | TEXT_PRESERVE_LIGATURES)``
Where in the documentation did you look at?

**Reply by AnweshGangula** [reply]
I would like to understand what that flag does.
https://pymupdf.readthedocs.io/en/latest/vars.html#TEXT_DEHYPHENATE
Above link mentions 
> Ignore hyphens at line ends and join with next line

Was wondering does this mean the quads will be joined into one list for hyphenated text!!

**Reply by JorjMcKie** [reply]
> Was wondering does this mean the quads will be joined into one list for hyphenated text!!

No, it means this:
Suppose your needle is "method" and your page contains three occurrences like this:
```text
some text on line blabla method more blabla meth-
od blablah on line 2 method.
```
You will get this list of 4 rectangles:
rect0 contains first "method"
rect1 contains "meth"
rect2 contains "od"
rect3 contains 3rd "method"

**Reply by JorjMcKie** [reply]
Sorry, I did not expect this could be misunderstood and will disambiguate the wording.

**Reply by AnweshGangula** [reply]
Got it. And you mentioned the search method already has this flag set. So are the 3 flags set by default for `page.search_for`?
```flags=TEXT_DEHYPHENATE | TEXT_PRESERVE_WHITESPACE | TEXT_PRESERVE_LIGATURES)```

**Reply by JorjMcKie** [reply]
Yes - I am trying to document everything with the default values shown.

**Reply by JorjMcKie** [reply]
All text methods are based on object `TextPage`. In its section you will also find other default flag combinations: they vary by `get_text(option)`.
